src/helper/remote_config/asm_data_listener.hpp

```diff
@@ -0,0 +1,28 @@
+// Unless explicitly stated otherwise all files in this repository are
+// dual-licensed under the Apache-2.0 License or BSD-3-Clause License.
+//
+// This product includes software developed at Datadog
+// (https://www.datadoghq.com/). Copyright 2021 Datadog, Inc.
+#pragma once
+
+#include "config.hpp"
+#include "listener.hpp"
+#include <optional>
+#include <rapidjson/document.h>
+
+namespace dds::remote_config {
+
+class asm_data_listener : public product_listener_base {
+public:
+    explicit asm_data_listener(
+        std::shared_ptr<dds::service_config> service_config)
```

Comment:
The `asm_data_listener` should be using the `engine` rather than the `service_config`. It should also be calling `update_rule_data` with the merged configuration (converted to `parameter`).

Reply:
Done